eva_sub_cli/validators/validator.py

```diff
@@ -445,6 +457,52 @@ def _convert_metadata_attribute(self, sheet, json_attribute, xls2json_conf):
             if attributes_dict[attribute] == json_attribute:
                 return attribute
 
+    def _collect_md5sum_to_metadata(self):
+        md5sum_file = resolve_single_file_path(os.path.join(self.output_dir, 'md5sums.txt'))
+        file_path_2_md5 = {}
+        file_name_2_md5 = {}
+        if md5sum_file:
+            with open(md5sum_file) as open_file:
+                for line in open_file:
+                    sp_line = line.split()
+                    md5sum = sp_line[0]
+                    vcf_file = sp_line[1]  # assuming no space in the vcf_file_name
+                    file_path_2_md5[vcf_file] = md5sum
+                    file_name_2_md5[os.path.basename(vcf_file)] = md5sum
+        if self.metadata_json_post_validation:
+            with open(self.metadata_json_post_validation) as open_file:
+                try:
+                    json_data = json.load(open_file)
+                    analysis_aliases = [a.get('analysisAlias') for a in json_data.get('analysis', [])]
+                    file_rows = []
+                    if len(analysis_aliases) == 1:
+                        analysis_alias = analysis_aliases[0]
+                        for vcf_file in self.vcf_files:
+                            validation_vcf_file = self._validation_file_path_for(vcf_file)
+                            file_rows.append({
+                                'analysisAlias': analysis_alias,
+                                'fileName': os.path.basename(vcf_file),
+                                'fileType': 'vcf',
+                                'md5': file_path_2_md5.get(validation_vcf_file) or
+                                       file_name_2_md5.get(os.path.basename(validation_vcf_file)) or ''
+                            })
+                    else:
+                        for file_dict in json_data.get('files', []):
+                            if file_dict.get('fileType') == 'vcf':
+                                if file_dict.get('fileName') in file_path_2_md5:
+                                    file_dict['md5'] = file_path_2_md5[file_dict.get('fileName')]
```

Comment:
If the user does provide an MD5 in their metadata, should we check that it matches the one that we computed? Or maybe we should just not ask for it, if we're going to overwrite it anyway? I think I would prefer one of those options, otherwise it seems weird to ask for a checksum (even optionally) and then not check it.

Reply:
That's a good point. It's a bit complicated because we still need to request it in the (hypothetical) case the JSON is provided directly to the API.

Reply:
My take on this now is to not request it in the spreadsheet but allow it in the JSON schema. We could document that the md5 will be calculated and inserted by the validation stage.